src/main/java/org/tron/common/runtime/vm/program/InternalTransaction.java

```diff
@@ -115,87 +137,67 @@ public String getNote() {
     return parentHash;
   }
 
-//    @Override
-//    public byte[] getEncoded() {
-//        if (protoEncoded == null) {
-//            /*
-//            this.protoEncoded = RLP.encodeList(
-//                    RLP.encodeElement(isEmptyNonce ? null : nonce),
-//                    RLP.encodeElement(this.parentHash),
-//                    RLP.encodeElement(getSender()),
-//                    RLP.encodeElement(getReceiveAddress()),
-//                    RLP.encodeElement(getValue()),
-//                    RLP.encodeElement(getGasPrice()),
-//                    RLP.encodeElement(getGasLimit()),
-//                    RLP.encodeElement(getData()),
-//                    RLP.encodeString(this.note),
-//                    encodeInt(this.deep),
-//                    encodeInt(this.index),
-//                    encodeInt(this.rejected ? 1 : 0)
-//            );
-//            */
-//        }
-//
-//        return protoEncoded;
-//    }
+  public long getValue() {
+    protoParse();
+    return value;
+  }
 
-  @Override
-  public byte[] getEncodedRaw() {
-    return getEncoded();
-  }
-
-//    @Override
-//    public synchronized void protoParse() {
-//        if (parsed) return;
-//        /*
-//        RLPList decodedTxList = RLP.decode2(rlpEncoded);
-//        RLPList transaction = (RLPList) decodedTxList.get(0);
-//
-//        this.parentHash = transaction.get(1).getRLPData();
-//        this.sendAddress = transaction.get(2).getRLPData();
-//        setReceiveAddress(transaction.get(3).getRLPData());
-//        setValue(transaction.get(4).getRLPData());
-//        setGasPrice(transaction.get(5).getRLPData());
-//        setGasLimit(transaction.get(6).getRLPData());
-//        setData(transaction.get(7).getRLPData());
-//        this.note = new String(transaction.get(8).getRLPData());
-//        this.deep = decodeInt(transaction.get(9).getRLPData());
-//        this.index = decodeInt(transaction.get(10).getRLPData());
-//        this.rejected = decodeInt(transaction.get(11).getRLPData()) == 1;
-//        */
-//        this.parsed = true;
-//    }
-
-
-  private static byte[] intToBytes(int value) {
-    return ByteBuffer.allocate(Integer.SIZE / Byte.SIZE)
-        .order(ByteOrder.LITTLE_ENDIAN)
-        .putInt(value)
-        .array();
-  }
-
-  private static int bytesToInt(byte[] bytes) {
-    return isEmpty(bytes) ? 0 : ByteBuffer.wrap(bytes).order(ByteOrder.LITTLE_ENDIAN).getInt();
-  }
-
-    /*
-    private static byte[] encodeInt(int value) {
-        return RLP.encodeElement(intToBytes(value));
-    }
-    */
+  public byte[] getData() {
+    protoParse();
+    return data;
```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: Return a copy of "data". ([squid:S2384](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2384))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1220&issues=AWT01HPBvlEfaeH0aL_d&open=AWT01HPBvlEfaeH0aL_d)

[ISSUE_KEY]: AWT01HPBvlEfaeH0aL_d